Sources/Async/Socket/SocketSource.swift

```diff
@@ -113,14 +120,37 @@ public final class SocketSource<Socket>: OutputStream
             return
         }
 
-        guard let readSource = self.readSource else {
-            fatalError("SocketSource readSource illegally nil during signal.")
+        guard downstreamIsReady else {
+            // downstream is not ready for data yet
+            excessSignalCount += 1
```

Comment:
`excessSignalCount = excessSignalCount &+ 1`

I prefer using `&+` in low level libraries (when safety is guaranteed). Since the check one line beneath basically guarantees no overflow will happen I think this worth it.

`&+` and `&-` remove the over/underflow checks from the single operation, which saves about 2-4x the performance.